homeassistant/components/iqvia/__init__.py

```diff
@@ -143,42 +141,100 @@
 class IQVIAData:
     """Define a data object to retrieve info from IQVIA."""
 
-    def __init__(self, client):
+    def __init__(self, hass, client):
         """Initialize."""
+        self._async_cancel_time_interval_listener = None
         self._client = client
+        self._hass = hass
         self.data = {}
         self.zip_code = client.zip_code
 
-        self.fetchers = Registry()
-        self.fetchers.register(TYPE_ALLERGY_FORECAST)(self._client.allergens.extended)
-        self.fetchers.register(TYPE_ALLERGY_OUTLOOK)(self._client.allergens.outlook)
-        self.fetchers.register(TYPE_ALLERGY_INDEX)(self._client.allergens.current)
-        self.fetchers.register(TYPE_ASTHMA_FORECAST)(self._client.asthma.extended)
-        self.fetchers.register(TYPE_ASTHMA_INDEX)(self._client.asthma.current)
-        self.fetchers.register(TYPE_DISEASE_FORECAST)(self._client.disease.extended)
-        self.fetchers.register(TYPE_DISEASE_INDEX)(self._client.disease.current)
+        # Define a registry to hold onto API coroutines:
+        self.api_coros = Registry()
+        self.api_coros.register(TYPE_ALLERGY_FORECAST)(client.allergens.extended)
+        self.api_coros.register(TYPE_ALLERGY_INDEX)(client.allergens.current)
+        self.api_coros.register(TYPE_ALLERGY_OUTLOOK)(client.allergens.outlook)
+        self.api_coros.register(TYPE_ASTHMA_FORECAST)(client.asthma.extended)
+        self.api_coros.register(TYPE_ASTHMA_INDEX)(client.asthma.current)
+        self.api_coros.register(TYPE_DISEASE_FORECAST)(client.disease.extended)
+        self.api_coros.register(TYPE_DISEASE_INDEX)(client.disease.current)
```

Comment:
Interesting choice to use this decorator. This is equivalent to
```suggestion
        self.api_coros = {
            TYPE_ALLERGY_FORECAST: client.allergens.extended,
            TYPE_ALLERGY_INDEX: client.allergens.current,
            TYPE_ALLERGY_OUTLOOK: client.allergens.outlook,
            TYPE_ASTHMA_FORECAST: client.asthma.extended,
            TYPE_ASTHMA_INDEX: client.asthma.current,
            TYPE_DISEASE_FORECAST: client.disease.extended,
            TYPE_DISEASE_INDEX: client.disease.current,
        }
```